src/contracts/Airdrop.h

```diff
@@ -0,0 +1,226 @@
+using namespace QPI;
+
+#define AIRDROP_START_SUCCESS 0
+#define AIRDROP_INSUFFICIENT_FUND 1
+#define AIRDROP_STARTED 2
+#define AIRDROP_SEND_FUND 3
+#define AIRDROP_START_ERROR 4
+#define AIRDROP_INCORRECT_TIME 5
+#define AIRDROP_DISTRIBUTE_SUCCESS 6
+#define AIRDROP_DISTRIBUTE_FAILD 7
+#define AIRDROP_PERIOD_LIMITED 8
+#define AIRDROP_FUND_LIMITED 9
+#define AIRDROP_INSUFFICIENT_TOKEN 10
+
+struct AirdropLogger
+{
+    id src;
+    id dst;
+    sint64 amt;
+    uint32 logtype;
+    char _terminator;
+};
+
+struct AIRDROP2
+{
+};
+
+struct AIRDROP
+{
+private:
+    struct _AssetOrder
+	{
+		id entity;
+		sint64 numberOfShares;
+	};
+	collection<_AssetOrder, 2097152 * X_MULTIPLIER> _assetOrders;
+
+	struct _EntityOrder
+	{
+		id issuer;
+		uint64 assetName;
+		sint64 numberOfShares;
+	};
+	collection<_EntityOrder, 2097152 * X_MULTIPLIER> _entityOrders;
+
+    id contractOwner;             // address of owner
+    sint64 TotalAmountForAirdrop; // amount of Airdrop
+    uint64 start_time;            // start time of Airdrop
+    uint64 end_time;              // end time of Airdrop
+    uint64 _burnedAmount;
+    AirdropLogger logger;
+    uint32 _airdropStartFee;// Amount of qus
+	uint32 _transferFee; // Amount of qus
+    sint64 _elementIndex, _elementIndex2;
+	id _issuerAndAssetName;
+	_AssetOrder _assetOrder;
+	_EntityOrder _entityOrder;
```

Comment:
For new contracts, I recommend to exclude temporary variables from the state completely, including logger, element indices etc. It is now possible to define a struct named "[FUNC_OR_PROC_NAME]_locals" that will be made available as the variable "locals" in the function/procedure (initialized with zeros) if you append "_WITH_LOCALS" to the definition of the procedure/function. For an example, see https://github.com/qubic/core/blob/main/src/contracts/Qx.h#L223-L245 
There is no example with procedures yet, because I did not want to change the states of existing contracts more than necessary.